dll/win32/uxtheme/system.c

```diff
@@ -1153,32 +1162,66 @@ HRESULT WINAPI CloseThemeFile(HTHEMEFILE hThemeFile)
  *
  * PARAMS
  *     hThemeFile           Handle to theme file
- *     unknown              See notes
+ *     Flags                Unknown flag bits
  *     hWnd                 Window requesting the theme change
  *
  * RETURNS
  *     Success: S_OK
  *     Failure: HRESULT error-code
- *
- * NOTES
- * I'm not sure what the second parameter is (the datatype is likely wrong), other then this:
- * Under XP if I pass
- * char b[] = "";
- *   the theme is applied with the screen redrawing really badly (flickers)
- * char b[] = "\0"; where \0 can be one or more of any character, makes no difference
- *   the theme is applied smoothly (screen does not flicker)
- * char *b = "\0" or NULL; where \0 can be zero or more of any character, makes no difference
- *   the function fails returning invalid parameter... very strange
  */
+#ifdef __REACTOS__
+HRESULT WINAPI ApplyTheme(HTHEMEFILE hThemeFile, UINT Flags, HWND hWnd)
+#else
 HRESULT WINAPI ApplyTheme(HTHEMEFILE hThemeFile, char *unknown, HWND hWnd)
+#endif
 {
     HRESULT hr;
+#ifdef __REACTOS__
+    TRACE("(%p,%#x,%p)\n", hThemeFile, Flags, hWnd);
+#else
     TRACE("(%p,%s,%p)\n", hThemeFile, unknown, hWnd);
+#endif
     hr = UXTHEME_ApplyTheme(hThemeFile);
     UXTHEME_broadcast_theme_changed (NULL, (g_ActiveThemeFile != NULL));
     return hr;
 }
 
+#ifdef __REACTOS__
+/**********************************************************************
+ *      SetSystemVisualStyle                              (UXTHEME.65)
+ */
+HRESULT WINAPI SetSystemVisualStyle(PCWSTR pszStyleFile, PCWSTR pszColor, PCWSTR pszSize, UINT Flags)
+{
+    HTHEMEFILE hThemeFile = NULL;
+    HRESULT hr;
+
+    if (pszStyleFile)
+    {
+        WCHAR szColor[256], szSize[256];
+        if (!pszColor || !*pszColor)
+        {
+            GetThemeDefaults(pszStyleFile, szColor, 256, NULL, 0);
+            pszColor = szColor;
+        }
+        if (!pszSize || !*pszSize)
+        {
+            GetThemeDefaults(pszStyleFile, NULL, 0, szSize, 256);
+            pszSize = szSize;
+        }
+
+        hr = OpenThemeFile(pszStyleFile, pszColor, pszSize, &hThemeFile, 0);
```

Comment:
This then could become an `MSSTYLES_OpenThemeFile` call instead (and the other two functions below become calls to their corresponding `MSSTYLES_` ones)